pkg/cloudprovider/providers/vsphere/vclib/datacenter.go

```diff
@@ -138,14 +138,8 @@ func (dc *Datacenter) GetDatastoreByPath(ctx context.Context, vmDiskPath string)
 		glog.Errorf("Failed to parse vmDiskPath: %s", vmDiskPath)
 		return nil, errors.New("Failed to parse vmDiskPath")
 	}
-	finder := getFinder(dc)
-	ds, err := finder.Datastore(ctx, datastorePathObj.Datastore)
-	if err != nil {
-		glog.Errorf("Failed while searching for datastore: %s. err: %+v", datastorePathObj.Datastore, err)
-		return nil, err
-	}
-	datastore := Datastore{ds, dc}
-	return &datastore, nil
+
+	return dc.GetDatastoreByName(ctx, datastorePathObj.Datastore)
```

Comment:
this does not belong into this PR, does it?

Reply:
Also some other changes.

Reply:
Thoes code's logic the same as dc.GetDatastoreByName(), I think we need reuse it instead of rewrite it. Sorry, i will make this PR more cleanly.